From 5236d3dfc1e5b1ab835f4f33ece6654cf4f55ba1 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:07:22 -0400
Subject: [tools] perf: Move ACCESS_ONCE from perf.h header

Message-id: <1410181752-25631-198-git-send-email-jolsa@redhat.com>
Patchwork-id: 91815
O-Subject: [PATCH RHEL7.1 BZ1134356 197/307] perf tools: Move ACCESS_ONCE from perf.h header
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 73a31b7c6a85e2f033524058bad5bce900e4f2ed
Author: Jiri Olsa <jolsa@kernel.org>
Date: Mon May 5 13:16:41 2014 +0200

description
===========
Into tools/include/linux/compiler.h header.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/include/linux/compiler.h b/tools/include/linux/compiler.h
index fbc6665..88461f0 100644
--- a/tools/include/linux/compiler.h
+++ b/tools/include/linux/compiler.h
@@ -35,4 +35,6 @@
 # define unlikely(x)  __builtin_expect(!!(x), 0)
 #endif
 
+#define ACCESS_ONCE(x) (*(volatile typeof(x) *)&(x))
+
 #endif /* _TOOLS_LINUX_COMPILER_H */
diff --git a/tools/perf/perf.h b/tools/perf/perf.h
index 813571b..510c65f 100644
--- a/tools/perf/perf.h
+++ b/tools/perf/perf.h
@@ -14,8 +14,6 @@ void test_attr__open(struct perf_event_attr *attr, pid_t pid, int cpu,
 #define HAVE_ATTR_TEST
 #include "perf-sys.h"
 
-#define ACCESS_ONCE(x) (*(volatile typeof(x) *)&(x))
-
 #ifndef NSEC_PER_SEC
 # define NSEC_PER_SEC   1000000000ULL
 #endif
-- 
1.7.1